packages/server/src/app/db/notifications/unpublished-notifications.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { performance } from 'perf_hooks';
+
+import { NotificationSubscriptionDocument } from './notification-subscriptions.schema';
+import { NotificationDocument } from './notifications.schema';
+import { UnpublishedNotificationDocument } from './unpublished-notifications.schema';
+
+@Injectable()
+export class UnpublishedNotificationsService {
+    constructor(@InjectModel('UnpublishedNotification') private readonly unpubNotifModel: Model<UnpublishedNotificationDocument>,
+        @InjectModel('NotificationSubscription') private readonly subscriptionModel: Model<NotificationSubscriptionDocument>,
+        @InjectModel('Notification') private readonly notificationModel: Model<NotificationDocument>) {
+
+        // Set up per-tick callback
+        setImmediate(async () => await this.processNotifications(100));
+    }
+
+    async processNotifications(numToProcess: number) { 
```

Comment:
This whole service is really just an excuse for this one eternally-looping method. That's one reason I'm thinking of folding it into the `NotificationsService`.